src/Pool.sol

```diff
@@ -345,6 +363,50 @@ abstract contract Pool {
         receiver.nextCollectedCycle = cycle;
     }
 
+    /// @notice Gives funds to the receiver.
+    /// The receiver can collect them immediately.
+    /// @param giverAddr The address of the giver
+    /// @param receiverAddr The receiver
+    /// @param amt The given amount
+    function _giveInternal(
+        address giverAddr,
```

Comment:
The `giverAddr` always needs to be msg.sender otherwise it is a huge security risk. I could just use the address of another pool user to give his funds. 

I think we can remove it and just use `msg.sender`

Reply:
`giverAddr` being `msg.sender` is enforced by `EthPool` and `ERC20Pool`, `_giveInternal` is an internal tool and nobody can call it freely.

This function follows the pattern of `Pool` never touching `msg.sender`. For example `_updateSenderInternal` too accepts the sender address and leaves decision of who it is to the higher-level implementations. We've introduced that pattern back in the `NFTPool` days and since then (AFAIR) we've had a consensus that it's a clean separation of concerns.

Reply:
True. We basically assume correct usage on the high level. It is anyway the same for updateSender as well. So i guess it is fine.

Reply:
I think the name `_giveFromAnyGiver` started this thought train. Because you can only give from msg.sender but I guess you indentation was any address can give but obviously only from their address :)

Reply:
It's an unfortunate attempt to follow `_updateSenderInternal` - `_updateSubSenderInternal` - `_updateAnySender` :grimacing: